Change the build configuration to decrease size and attack area

This is made up of:
* Removing support for various block device formats.
* Removing support for VFIO.
* Disabling many devices which are not needed/used.

diff --git a/block/Makefile.objs b/block/Makefile.objs
index 2aaede4..9907714 100644
--- a/block/Makefile.objs
+++ b/block/Makefile.objs
@@ -1,10 +1,12 @@
-block-obj-y += raw-format.o qcow.o vdi.o vmdk.o cloop.o bochs.o vpc.o vvfat.o dmg.o
+block-obj-y += raw-format.o qcow.o vdi.o
+block-obj-$(call lnot,$(CONFIG_XEN)) += vmdk.o cloop.o bochs.o vpc.o vvfat.o dmg.o
 block-obj-y += qcow2.o qcow2-refcount.o qcow2-cluster.o qcow2-snapshot.o qcow2-cache.o qcow2-bitmap.o
 block-obj-y += qed.o qed-l2-cache.o qed-table.o qed-cluster.o
 block-obj-y += qed-check.o
-block-obj-y += vhdx.o vhdx-endian.o vhdx-log.o
+block-obj-$(call lnot,$(CONFIG_XEN)) += vhdx.o vhdx-endian.o vhdx-log.o
 block-obj-y += quorum.o
-block-obj-y += parallels.o blkdebug.o blkverify.o blkreplay.o
+block-obj-y += blkdebug.o blkverify.o blkreplay.o
+block-obj-$(call lnot,$(CONFIG_XEN)) += parallels.o
 block-obj-y += block-backend.o snapshot.o qapi.o
 block-obj-$(CONFIG_WIN32) += file-win32.o win32-aio.o
 block-obj-$(CONFIG_POSIX) += file-posix.o
@@ -12,7 +14,7 @@ block-obj-$(CONFIG_LINUX_AIO) += linux-aio.o
 block-obj-y += null.o mirror.o commit.o io.o
 block-obj-y += throttle-groups.o
 
-block-obj-y += nbd.o nbd-client.o sheepdog.o
+block-obj-$(call lnot,$(CONFIG_XEN)) += nbd.o nbd-client.o sheepdog.o
 block-obj-$(CONFIG_LIBISCSI) += iscsi.o
 block-obj-$(if $(CONFIG_LIBISCSI),y,n) += iscsi-opts.o
 block-obj-$(CONFIG_LIBNFS) += nfs.o
diff --git a/default-configs/i386-softmmu.mak b/default-configs/i386-softmmu.mak
index d2ab2f6..0d73182 100644
--- a/default-configs/i386-softmmu.mak
+++ b/default-configs/i386-softmmu.mak
@@ -6,19 +6,19 @@ include usb.mak
 CONFIG_QXL=$(CONFIG_SPICE)
 CONFIG_VGA_ISA=y
 CONFIG_VGA_CIRRUS=y
-CONFIG_VMWARE_VGA=y
-CONFIG_VIRTIO_VGA=y
-CONFIG_VMMOUSE=y
-CONFIG_IPMI=y
-CONFIG_IPMI_LOCAL=y
-CONFIG_IPMI_EXTERN=y
-CONFIG_ISA_IPMI_KCS=y
-CONFIG_ISA_IPMI_BT=y
+CONFIG_VMWARE_VGA=n
+CONFIG_VIRTIO_VGA=n
+CONFIG_VMMOUSE=n
+CONFIG_IPMI=n
+CONFIG_IPMI_LOCAL=n
+CONFIG_IPMI_EXTERN=n
+CONFIG_ISA_IPMI_KCS=n
+CONFIG_ISA_IPMI_BT=n
 CONFIG_SERIAL=y
 CONFIG_SERIAL_ISA=y
 CONFIG_PARALLEL=y
 CONFIG_I8254=y
-CONFIG_PCSPK=y
+CONFIG_PCSPK=n
 CONFIG_PCKBD=y
 CONFIG_FDC=y
 CONFIG_ACPI=y
@@ -30,21 +30,21 @@ CONFIG_APM=y
 CONFIG_I8257=y
 CONFIG_IDE_ISA=y
 CONFIG_IDE_PIIX=y
-CONFIG_NE2000_ISA=y
-CONFIG_HPET=y
-CONFIG_APPLESMC=y
+CONFIG_NE2000_ISA=n
+CONFIG_HPET=n
+CONFIG_APPLESMC=n
 CONFIG_I8259=y
 CONFIG_PFLASH_CFI01=y
 CONFIG_TPM_TIS=$(CONFIG_TPM)
 CONFIG_MC146818RTC=y
 CONFIG_PCI_PIIX=y
-CONFIG_WDT_IB700=y
-CONFIG_ISA_DEBUG=y
-CONFIG_ISA_TESTDEV=y
-CONFIG_VMPORT=y
-CONFIG_SGA=y
+CONFIG_WDT_IB700=n
+CONFIG_ISA_DEBUG=n
+CONFIG_ISA_TESTDEV=n
+CONFIG_VMPORT=n
+CONFIG_SGA=n
 CONFIG_LPC_ICH9=y
-CONFIG_PCI_Q35=y
+CONFIG_PCI_Q35=n
 CONFIG_APIC=y
 CONFIG_IOAPIC=y
 CONFIG_PVPANIC=y
@@ -52,10 +52,10 @@ CONFIG_MEM_HOTPLUG=y
 CONFIG_NVDIMM=y
 CONFIG_ACPI_NVDIMM=y
 CONFIG_PCIE_PORT=y
-CONFIG_XIO3130=y
-CONFIG_IOH3420=y
-CONFIG_I82801B11=y
+CONFIG_XIO3130=n
+CONFIG_IOH3420=n
+CONFIG_I82801B11=n
 CONFIG_SMBIOS=y
 CONFIG_HYPERV_TESTDEV=$(CONFIG_KVM)
-CONFIG_PXB=y
+CONFIG_PXB=n
 CONFIG_ACPI_VMGENID=y
diff --git a/default-configs/pci.mak b/default-configs/pci.mak
index a758630..106ee63 100644
--- a/default-configs/pci.mak
+++ b/default-configs/pci.mak
@@ -1,46 +1,46 @@
 CONFIG_PCI=y
 # For now, CONFIG_IDE_CORE requires ISA, so we enable it here
 CONFIG_ISA_BUS=y
-CONFIG_VIRTIO_PCI=y
-CONFIG_VIRTIO=y
+CONFIG_VIRTIO_PCI=n
+CONFIG_VIRTIO=n
 CONFIG_USB_UHCI=y
 CONFIG_USB_OHCI=y
 CONFIG_USB_EHCI=y
-CONFIG_USB_XHCI=y
-CONFIG_USB_XHCI_NEC=y
-CONFIG_NE2000_PCI=y
-CONFIG_EEPRO100_PCI=y
-CONFIG_PCNET_PCI=y
-CONFIG_PCNET_COMMON=y
-CONFIG_AC97=y
-CONFIG_HDA=y
-CONFIG_ES1370=y
-CONFIG_LSI_SCSI_PCI=y
-CONFIG_VMW_PVSCSI_SCSI_PCI=y
-CONFIG_MEGASAS_SCSI_PCI=y
-CONFIG_MPTSAS_SCSI_PCI=y
+CONFIG_USB_XHCI=n
+CONFIG_USB_XHCI_NEC=n
+CONFIG_NE2000_PCI=n
+CONFIG_EEPRO100_PCI=n
+CONFIG_PCNET_PCI=n
+CONFIG_PCNET_COMMON=n
+CONFIG_AC97=n
+CONFIG_HDA=n
+CONFIG_ES1370=n
+CONFIG_LSI_SCSI_PCI=n
+CONFIG_VMW_PVSCSI_SCSI_PCI=n
+CONFIG_MEGASAS_SCSI_PCI=n
+CONFIG_MPTSAS_SCSI_PCI=n
 CONFIG_RTL8139_PCI=y
-CONFIG_E1000_PCI=y
-CONFIG_E1000E_PCI=y
-CONFIG_VMXNET3_PCI=y
+CONFIG_E1000_PCI=n
+CONFIG_E1000E_PCI=n
+CONFIG_VMXNET3_PCI=n
 CONFIG_IDE_CORE=y
 CONFIG_IDE_QDEV=y
 CONFIG_IDE_PCI=y
 CONFIG_AHCI=y
-CONFIG_ESP=y
-CONFIG_ESP_PCI=y
+CONFIG_ESP=n
+CONFIG_ESP_PCI=n
 CONFIG_SERIAL=y
 CONFIG_SERIAL_ISA=y
-CONFIG_SERIAL_PCI=y
-CONFIG_IPACK=y
-CONFIG_WDT_IB6300ESB=y
-CONFIG_PCI_TESTDEV=y
-CONFIG_NVME_PCI=y
-CONFIG_SD=y
-CONFIG_SDHCI=y
-CONFIG_EDU=y
+CONFIG_SERIAL_PCI=n
+CONFIG_IPACK=n
+CONFIG_WDT_IB6300ESB=n
+CONFIG_PCI_TESTDEV=n
+CONFIG_NVME_PCI=n
+CONFIG_SD=n
+CONFIG_SDHCI=n
+CONFIG_EDU=n
 CONFIG_VGA=y
 CONFIG_VGA_PCI=y
-CONFIG_IVSHMEM_DEVICE=$(CONFIG_IVSHMEM)
-CONFIG_ROCKER=y
+CONFIG_IVSHMEM_DEVICE=n
+CONFIG_ROCKER=n
 CONFIG_VHOST_USER_SCSI=$(and $(CONFIG_VHOST_USER),$(CONFIG_LINUX))
diff --git a/default-configs/sound.mak b/default-configs/sound.mak
index 4f22c34..55f64b8 100644
--- a/default-configs/sound.mak
+++ b/default-configs/sound.mak
@@ -1,4 +1,4 @@
-CONFIG_SB16=y
-CONFIG_ADLIB=y
-CONFIG_GUS=y
-CONFIG_CS4231A=y
+CONFIG_SB16=n
+CONFIG_ADLIB=n
+CONFIG_GUS=n
+CONFIG_CS4231A=n
diff --git a/default-configs/usb.mak b/default-configs/usb.mak
index f4b8568..7e52934 100644
--- a/default-configs/usb.mak
+++ b/default-configs/usb.mak
@@ -1,10 +1,10 @@
 CONFIG_USB=y
 CONFIG_USB_TABLET_WACOM=y
-CONFIG_USB_STORAGE_BOT=y
-CONFIG_USB_STORAGE_UAS=y
-CONFIG_USB_STORAGE_MTP=y
-CONFIG_USB_SMARTCARD=y
-CONFIG_USB_AUDIO=y
-CONFIG_USB_SERIAL=y
-CONFIG_USB_NETWORK=y
-CONFIG_USB_BLUETOOTH=y
+CONFIG_USB_STORAGE_BOT=n
+CONFIG_USB_STORAGE_UAS=n
+CONFIG_USB_STORAGE_MTP=n
+CONFIG_USB_SMARTCARD=n
+CONFIG_USB_AUDIO=n
+CONFIG_USB_SERIAL=n
+CONFIG_USB_NETWORK=n
+CONFIG_USB_BLUETOOTH=n
diff --git a/hw/vfio/Makefile.objs b/hw/vfio/Makefile.objs
index c3ab909..9e9fe59 100644
--- a/hw/vfio/Makefile.objs
+++ b/hw/vfio/Makefile.objs
@@ -1,4 +1,4 @@
-ifeq ($(CONFIG_LINUX), y)
+ifeq ($(CONFIG_LINUX)$(CONFIG_XEN), yn)
 obj-$(CONFIG_SOFTMMU) += common.o
 obj-$(CONFIG_PCI) += pci.o pci-quirks.o
 obj-$(CONFIG_VFIO_CCW) += ccw.o
diff --git a/hw/virtio/Makefile.objs b/hw/virtio/Makefile.objs
index 765d363..015fe60 100644
--- a/hw/virtio/Makefile.objs
+++ b/hw/virtio/Makefile.objs
@@ -11,5 +11,5 @@ obj-y += virtio-crypto.o
 obj-$(CONFIG_VIRTIO_PCI) += virtio-crypto-pci.o
 endif
 
-common-obj-$(call lnot,$(CONFIG_LINUX)) += vhost-stub.o
+common-obj-$(call lnot,$(CONFIG_VIRTIO)) += vhost-stub.o
 common-obj-$(CONFIG_ALL) += vhost-stub.o
